src/query/query_executor.c

```diff
@@ -7916,6 +7957,13 @@ qexec_intprt_fnc (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_s
 	  qexec_clear_all_lists (thread_p, xasl);
 	}
 
+      if (max_recursive_iterations_reached)
+	{
+	  xb_scan = S_END;
+	  er_set (ER_NOTIFICATION_SEVERITY, ARG_FILE_LINE, ER_CTE_MAX_RECURSION_REACHED, 1, xasl->max_iterations);
+	  break;
```

Comment:
This was a silly :)
I expect to return error.
Oracle detects the cycle and returns error.
I fixed the code to return error.